mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -4875,23 +4892,39 @@ struct TransformRewritePattern : public OpRewritePattern<miopen::TransformOp> {
                   AffineMapAttr::get(outputType.getAffineMaps()[0]));
 
               auto existingDomain = dictAttr.get("domain").cast<ArrayAttr>();
+              auto existingLimits = dictAttr.get("limits");
 
-              augmentedArrayAttr.push_back(b.getDictionaryAttr(
-                  {b.getNamedAttr("operand",
-                                  b.getI32IntegerAttr(userOperandIndex)),
-                   b.getNamedAttr("transforms",
-                                  b.getArrayAttr(augmentedTransforms)),
-                   b.getNamedAttr("domain", existingDomain)}));
+              llvm::SmallVector<NamedAttribute, 4> arrayAttr{
+                  b.getNamedAttr("operand",
+                                 b.getI32IntegerAttr(userOperandIndex)),
+                  b.getNamedAttr("transforms",
+                                 b.getArrayAttr(augmentedTransforms)),
+                  b.getNamedAttr("domain", existingDomain)};
+
+              if (existingLimits)
+                arrayAttr.push_back(
+                    b.getNamedAttr("limits", existingLimits.cast<ArrayAttr>()));
```

Comment:
what would happen when the dimension number and ordering gets changed?

Reply:
Only the most original dimensions can be recorded